### PR TITLE
Make dependency of VISSL optional

### PR DESCRIPTION
**Pull request by ar90n**
## What does this PR do?
This PR is motivated to achieve the second change in [this comment](https://github.com/PyTorchLightning/lightning-flash/issues/1031#issuecomment-1083103525). And this PR contains the following modifications.

* Make `training_strategy`, `head`, `pretraining_transform` in the arguments of `__init__` of ImageEmbedder optional
* Add `DefaultAdapter` to the Embedding task which is inspired by the Image Classification task's one.
* Fix a test about integration with Fiftyone to work correctly
* Add test cases for only embedding without VISSL

Part of https://github.com/PyTorchLightning/lightning-flash/issues/1031

This PR depends on #1264. So this PR won't be changed to Ready for review until #1264 is merged.

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/image/embedding/model.py [hunk 1]
```diff
@@ -77,9 +77,9 @@ class ImageEmbedder(AdapterTask):
 
     def __init__(
         self,
-        training_strategy: str,
-        head: str,
-        pretraining_transform: str,
+        training_strategy: Optional[str] = "default",
```
Comment (ethanwharris) [inline flash/image/embedding/model.py:80]: No need for the optional typing as I guess we don't support `None` here:
```suggestion
        training_strategy: str = "default",
```

File: flash/image/embedding/strategies/default.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        head: torch.nn.Module,
+        **kwargs,
+    ) -> Adapter:
+        adapter = cls(backbone, head)
+        adapter.__dict__["_task"] = task
+        return adapter
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DataKeys.INPUT], batch[DataKeys.TARGET])
+        return Task.training_step(self._task, batch, batch_idx)
```
Comment (ethanwharris) [inline flash/image/embedding/strategies/default.py:41]: Let's instead raise an error here as we only support training the embedder if using the VISSL integration. Can do something similar to what we have here: https://github.com/PyTorchLightning/lightning-flash/blob/167f5e73fe28630830e15ed40502979a29d38a68/flash/text/embedding/model.py#L81

File: flash/image/embedding/strategies/default.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    def forward(self, x) -> torch.Tensor:
+        # TODO: Resolve this hack
+        if x.dim() == 3:
+            x = x.unsqueeze(0)
+        return self.backbone(x)
+
+
+def default(head: Optional[str] = None, loss_fn: Optional[str] = None, **kwargs):
+    if loss_fn in IMAGE_EMBEDDER_LOSS_FUNCTIONS:
+        loss_fn = IMAGE_EMBEDDER_LOSS_FUNCTIONS.get(loss_fn)(**kwargs)
```
Comment (ethanwharris) [inline flash/image/embedding/strategies/default.py:66]: I guess just warn here if the loss function or head aren't `None` (as we don't have any loss functions of heads that don't use VISSL)

**Review by ethanwharris — COMMENTED** [review]
@ar90n This is awesome!!! Just a few minor comments :smiley:

File: flash/image/embedding/strategies/default.py (hunk 2 above)
Comment (ar90n) [reply flash/image/embedding/strategies/default.py]: I fixed it. Please review again!

File: flash/image/embedding/strategies/default.py (hunk 3 above)
Comment (ar90n) [reply flash/image/embedding/strategies/default.py]: I agree with you. On second thought, it is enough that `default` return always `(None, None, [])`.  I think this makes `default` simple. What do you think about it?

And I modified this PR as I said above. If it is OK, please review it.
